### PR TITLE
globally shared dictionary components

### PR DESCRIPTION
**Pull request by mr-martian**
This PR creates and distributes an FST of punctuation and internet things (URLs, IPs, emails) based on the regexs in apertium-cat.

This FST can then be added to monolingual dictionaries using the new `lt-append`.

### CODE REVIEW

**Comment by TinoDidriksen** [comment]
Yes, but not in the apertium or lttoolbox or any C++ code repo.

**Comment by mr-martian** [comment]
so, a new `apertium-shared` or `apertium-data` repo?

**Comment by TinoDidriksen** [comment]
I looked into it back when I started packaging Giella stuff - and why giella-shared is packaged as giella-common (with an alias for shared). `shared` implies that it's a shared library as opposed to static library. `data` doesn't imply it's data other tools can use. `common` seems to be what people mostly use for shared common data.

But, as a data-only package I would bundle it for releases anyway, so the name is not terribly important. So `apertium-shared` is actually fine, since as a build dependency that's clearer to our users than -common or -data. And now that I do bundle giella-shared, the old decision for -common is similarly invalidated.

Long story short, yeah `apertium-shared` is good.

**Comment by mr-martian** [comment]
Transferred to https://github.com/apertium/apertium-shared/commit/45ab1ec6dd6cd94b53f86061ecf59e186d5f6004
